src/lib/rnp.cpp

```diff
@@ -4212,6 +4220,207 @@ rnp_key_get_uid_handle_at(rnp_key_handle_t key, size_t idx, rnp_uid_handle_t *ui
+{
+    if (!handle || !sig) {
+        return RNP_ERROR_NULL_POINTER;
+    }
+
+    pgp_key_t *key = get_key_prefer_public(handle);
+    if (!key) {
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+    return rnp_key_get_signature_at_for_uid(handle->ffi, key, idx, (uint32_t) -1, sig);
+}
+
+rnp_result_t
+rnp_uid_get_signature_count(rnp_uid_handle_t handle, size_t *count)
+{
+    if (!handle || !count) {
+        return RNP_ERROR_NULL_POINTER;
+    }
+    if (!handle->key) {
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+    return rnp_key_get_signature_count_for_uid(handle->key, count, handle->idx);
+}
+
+rnp_result_t
+rnp_uid_get_signature_at(rnp_uid_handle_t handle, size_t idx, rnp_signature_handle_t *sig)
+{
+    if (!handle || !sig) {
+        return RNP_ERROR_NULL_POINTER;
+    }
+    if (!handle->key) {
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+    return rnp_key_get_signature_at_for_uid(handle->ffi, handle->key, idx, handle->idx, sig);
+}
+
+rnp_result_t
+rnp_signature_get_alg(rnp_signature_handle_t handle, char **alg)
+{
+    if (!handle || !alg) {
+        return RNP_ERROR_NULL_POINTER;
+    }
+    if (!handle->sig) {
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+    const char *str = NULL;
+    ARRAY_LOOKUP_BY_ID(pubkey_alg_map, type, string, handle->sig->sig.palg, str);
+    if (!str) {
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+    char *algcp = strdup(str);
+    if (!algcp) {
+        return RNP_ERROR_OUT_OF_MEMORY;
+    }
+
+    *alg = algcp;
+    return RNP_SUCCESS;
+}
+
+rnp_result_t
+rnp_signature_get_hash_alg(rnp_signature_handle_t handle, char **alg)
+{
+    if (!handle || !alg) {
+        return RNP_ERROR_NULL_POINTER;
+    }
+    if (!handle->sig) {
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+    const char *str = NULL;
+    ARRAY_LOOKUP_BY_ID(hash_alg_map, type, string, handle->sig->sig.halg, str);
+    if (!str) {
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+    char *algcp = strdup(str);
+    if (!algcp) {
+        return RNP_ERROR_OUT_OF_MEMORY;
+    }
+
+    *alg = algcp;
+    return RNP_SUCCESS;
+}
+
+rnp_result_t
+rnp_signature_get_creation(rnp_signature_handle_t handle, uint32_t *create)
+{
+    if (!handle || !create) {
+        return RNP_ERROR_NULL_POINTER;
+    }
+    if (!handle->sig) {
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+    *create = signature_get_creation(&handle->sig->sig);
+    return RNP_SUCCESS;
+}
+
+rnp_result_t
+rnp_signature_get_keyid(rnp_signature_handle_t handle, char **result)
+{
+    if (!handle || !result) {
+        return RNP_ERROR_NULL_POINTER;
+    }
+    if (!handle->sig) {
+        return RNP_ERROR_BAD_PARAMETERS;
+    }
+    uint8_t keyid[PGP_KEY_ID_SIZE] = {0};
+    if (!signature_get_keyid(&handle->sig->sig, keyid)) {
+        *result = NULL;
+        return RNP_SUCCESS;
+    }
+
+    size_t hex_len = PGP_KEY_ID_SIZE * 2 + 1;
+    *result = (char *) malloc(hex_len);
+    if (*result == NULL) {
```

Comment:
Style-wise I would argue to use `if (!*result)` though I'm not sure we have an official guideline on it.

Reply:
I don’t think we have this specified in the guide, but I agree that your suggestion is more consistent (just a few lines above).

Reply:
Thanks, you are right. Will fix and merge this soon.